zetta_utils/mazepa_layer_processing/common/volumetric_apply_flow.py

```diff
@@ -50,7 +51,68 @@ def __call__(
 
 @mazepa.taskable_operation_cls
 @attrs.mutable
-class ReduceByWeightedSum:
+class ReduceOperation(ABC):
+    """Base class for Reduce operations, which combine values from different
+    chunks where they overlap."""
+
+    def __call__(
+        self,
+        src_idxs: List[VolumetricIndex],
+        src_layers: List[VolumetricBasedLayerProtocol],
+        red_idx: VolumetricIndex,
+        roi_idx: VolumetricIndex,
+        dst: VolumetricBasedLayerProtocol,
+        processing_blend_pad: Vec3D[int],
+    ) -> None:
+        pass
+
+
+@mazepa.taskable_operation_cls
+@attrs.mutable
```

Comment:
Should be `@attrs.frozen`.

Reply:
OK.  But next week maybe somebody can explain to me why the (pre-existing) Copy operation on line 39 and ReduceByWeightedSum on line 107 need @attrs.mutable, but these new ones don't.

Reply:
I think all of them should be frozen strictly speaking, but it doesn't matter too much. I'll just merge